### PR TITLE
fix(@angular/cli): ensure lint generates well-formed machine output

### LINKED ISSUE #5259: Duplicate Lint Errors
> Please provide us with the following information:
> ---------------------------------------------------------------

`OSX El Capitan Version 10.11.6`

```
@angular/cli: 1.0.0-rc.1
node: 6.9.5
os: darwin x64
```

1. Run `ng new lint-test`
2. Run `cd lint-test`
3. Run `ng g component components/test-lint`
4. Open `src/app/components/test-lint/test-lint.component.ts` and cause one lint error (e.g, use " instead of ')
5. Run `npm run lint`
6. Notice how the lint errors are logged twice

```
> linting-test@0.0.0 lint /path/to/lint-test
> ng lint

/path/to/lint-test/src/app/components/test-lint/test-lint.component.ts[4, 13]: " should be '
/path/to/lint-test/src/app/components/test-lint/test-lint.component.ts[4, 13]: " should be '
Lint errors found in the listed files.
```

### Mention any other details that might be useful.
Through a bit of trial and error it seems the issue only occurs on files that are nested inside subfolders like `components` and are imported to other nested files. In this case it happens out of the box because `ng new` generates a `test-lint.component.spec.ts` file for each component which has an `import { TestLintComponent } from './test-lint.component';` thus causing lint errors on the component to be reported twice.

A temporary fix in my particular case was to remove `"include": [ "**/*.spec.ts" ]` from `src/tsconfig.spec.json` since linting test specs was a priority to me.

It'd be nice to know more about the cause and if there is a better fix for this dilemma.

Thanks! ^^

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

### PR DESCRIPTION
**Pull request by clydin**
For machine readable output formats, `ng lint` was concatenating multiple _documents_ into a single file.  This resulted in invalid JSON, XML, etc.  The lint results are now aggregated and one full output is generated.

Fix #5259

### CODE REVIEW

File: packages/@angular/cli/tasks/lint.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { requireProjectModule } from '../utilities/require-project-module';
 import { CliConfig } from '../models/config';
 import { LintCommandOptions } from '../commands/lint';
 import { oneLine } from 'common-tags';
+// import * as tslint from 'tslint';
```
Comment (sumitarora) [inline packages/@angular/cli/tasks/lint.ts]: Please remove commented code

File: packages/@angular/cli/tasks/lint.ts [hunk 2]
```diff
@@ -56,19 +54,37 @@ export default Task.extend({
...
+      })
+      .reduce((total, current) => {
+        const failures = current.failures
+          .filter((cf: any) => !total.failures.some((ef: any) => ef.equals(cf)));
+        total.failures = total.failures.concat(...failures);
+
+        if (current.fixes) {
+          total.fixes = (total.fixes || []).concat(...current.fixes);
+        }
+        return total;
```
Comment (sumitarora) [inline packages/@angular/cli/tasks/lint.ts]: Can we just change it to 1 Liner `total.fixes = (total.fixes || []).concat(...(current.fixes || []));` 😉

Comment (clydin) [inline packages/@angular/cli/tasks/lint.ts]: Originally had a one-liner but some of the built-in formatters expect `fixes` to be _falsy_ if none were made.

File: packages/@angular/cli/tasks/lint.ts (hunk 1 above)
Comment (clydin) [inline packages/@angular/cli/tasks/lint.ts]: Done.

**Pull request closed without merging**